drivers/ieee802154/CMakeLists.txt

```diff
@@ -9,3 +9,8 @@ zephyr_sources_ifdef(CONFIG_IEEE802154_CC1200 ieee802154_cc1200.c)
 zephyr_sources_ifdef(CONFIG_IEEE802154_CC13XX_CC26XX ieee802154_cc13xx_cc26xx.c)
 zephyr_sources_ifdef(CONFIG_IEEE802154_RF2XX ieee802154_rf2xx.c)
 zephyr_sources_ifdef(CONFIG_IEEE802154_RF2XX ieee802154_rf2xx_iface.c)
+
+if(CONFIG_IEEE802154_GECKO)
+	zephyr_sources(ieee802154_gecko.c)
+	zephyr_link_libraries(RAIL_LIB)
```

Comment:
> 1. Is `zephyr_link_libraries(RAIL_LIB)` even necessary with the setting `APP_LINK_WITH_RAIL_LIB` already in place?

Actually, it is necessary. `APP_LINK_WITH_RAIL_LIB` makes rail_lib available to the application but not Zephyr itself.

> 2. If it is necessary, could we link with `RAIL_LIB`, if `SOC_GECKO_RAIL_LIB` is defined and move this line to e.g. soc folder? This would make it independent from IEEE802154, since with the rail lib, also other protocols can be implemented.

RAIL is a specialized library and my plan was to make it visible only to the drivers that are using it. I.e RAIL will not be used by gpio driver nor the logger subsystem. In this case BLE driver would also need to explicitly link with RAIL. All the option does is to add RAIL specific include paths via -I parameter when compiling the code. Limiting RAIL visibility and requiring drivers to explicitly link with it seemed like a cleaner approach. That said, I may have misunderstood intricacies of Zephyr build system since RAIL include paths are in fact visible when compiling gpio driver (but not when compiling ieee802154 driver). I will need to look a bit closer at the issue.

Reply:
Thank you for the explanations, sound reasonable.